### ISSUE TITLE
Test Cleanup

### ISSUE THREAD

**Issue by smudge202** [smudge202/compose#77]
//CC @herecydev @mattridgway @sblacker

The tests right now are simply a mess.

Some classes have dozens of regions, hundreds of class and interface declarations, and inconsistent naming.

I'm wondering if it would be an idea to follow a slightly more _spec_ based approach? i.e. Break out the existing test files into dozens of files that better describe the behavioural testing, more consistently.

I imagine we'll quickly agree a cleanup is necessary, but ideas on best approach for test scheme would be useful?

**Comment by sblackler** [comment]
Given the amount of traction this seems to be getting + views from streaming, I suggest this is quiet important to clean up now.

**Comment by smudge202** [comment]
Agreed. I need to dig into specflow / msspec and such. Probably be worth checking out the tests on Castle, Moq, etc to see how they overcome some of the mess.

Let's move this into the next drop/milestone to give it some priority.
